pkg/didcomm/protocol/route/service.go

```diff
@@ -205,33 +298,55 @@ func (s *Service) Name() string {
 	return Coordination
 }
 
-func (s *Service) handleRequest(msg service.DIDCommMsg, myDID, theirDID string) error {
+func (s *Service) handleInboundRequest(c *callback) error {
 	// unmarshal the payload
 	request := &Request{}
 
-	err := msg.Decode(request)
+	err := c.msg.Decode(request)
 	if err != nil {
 		return fmt.Errorf("route request message unmarshal : %w", err)
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/1133 Support to
-	//  add business logic for Route Request Approval
-
-	// create keys
-	_, sigPubKey, err := s.kms.CreateKeySet()
+	grant, err := outboundGrant(
+		c.msg.ID(),
+		c.options,
+		s.endpoint,
+		func() (string, error) {
+			_, key, er := s.kms.CreateKeySet()
+			return key, er
+		},
+	)
 	if err != nil {
-		return fmt.Errorf("failed to create keys : %w", err)
+		return fmt.Errorf("failed to handle inbound request : %w", err)
 	}
 
-	// send the grant response
+	return s.outbound.SendToDID(grant, c.myDID, c.theirDID)
+}
+
+func outboundGrant(
+	msgID string, opts *Options,
+	defaultEndpoint string, defaultKeys func() (string, error)) (*Grant, error) {
 	grant := &Grant{
+		ID:          msgID,
 		Type:        GrantMsgType,
-		ID:          msg.ID(),
-		Endpoint:    s.endpoint,
-		RoutingKeys: []string{sigPubKey},
+		Endpoint:    opts.ServiceEndpoint,
+		RoutingKeys: opts.RoutingKeys,
+	}
+
+	if grant.Endpoint == "" {
+		grant.Endpoint = defaultEndpoint
+	}
+
+	if len(grant.RoutingKeys) == 0 {
+		keys, err := defaultKeys()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create keys : %w", err)
+		}
+
+		grant.RoutingKeys = []string{keys}
```

Comment:
I think it should be `defaultKey` instead of `defaultKeys` and `key` instead of `keys`.

Reply:
done